### ISSUE TITLE
Block Editor - not enabling

### ISSUE THREAD

**Issue by skobylite** [Automattic/wp-desktop#813]
#### Steps to reproduce
1. Create draft post
2. Click "enable block editor" bottom right on document settings

#### What I expected
Block editor to be available

#### What happened instead
Prompt disappears, but block editor not enabled

#### App Version / OS version
5.0

#### Screenshot / Video
https://i.imgur.com/raIXnd5.jpg

![raIXnd5](https://user-images.githubusercontent.com/1119271/78392893-386e6000-75a6-11ea-8a1d-c49ffe130e0a.jpg)

**Comment by nsakaimbo** [comment]
Thanks for the report, @skobylite. 

By "enable block editor", are you referring to the button in the attached screenshot? 

<img width="278" alt="try-blocks" src="https://user-images.githubusercontent.com/8979548/77338157-52838500-6d00-11ea-8405-3454cfa5c746.png">

If so, I haven't been able to reproduce this bug yet. What platform are you on? (Windows/Mac/Linux?)

Would appreciate some further clarification if I'm not looking in the right place.

**Comment by skobylite** [comment]
Thanks @nsakaimbo - I clicked the 'enable block editor button' and it's no longer showing as an option, but the block editor is not enabled: https://imgur.com/a/4155UVc

**Comment by nsakaimbo** [comment]
@skobylite After activating the block editor, did you try quitting (`File -> Quit`) and re-starting the app? In my testing recently I found a bug in which a change of editor type (class vs block) won't be reflected until the application is re-started.

**Comment by skobylite** [comment]
Thanks nsaikaimbo - I've just tried restarting the app, uninstalling and reinstalling, and trying on different sites with no success.
Do you have any other suggestions on how to activate the block editor?

**Comment by skobylite** [comment]
Just to check - should block editor work in the mac app for both .org and .com wordpress sites?

**Comment by nsakaimbo** [comment]
@skobylite Can you confirm whether you can use the block editor on the web and not in the app? Verifying whether or not the behavior is consistent outside of the app would be helpful in isolating where the problem is.

The application can be used for .org sites, but only if the sites have Jetpack connected.

**Comment by skobylite** [comment]
I normally use the block editor on the web, and I've got jetpack connected, so the problem seems to be specific to the app

**Comment by designsimply** [comment]
@nsakaimbo I tried testing this and I don't see a way to enable the block editor at all in my test Jetpack sites on the app and if I do somehow get that working then the app still won't open the block editor but instead sends me to the web which looks to me like a regressions of https://github.com/Automattic/wp-desktop/issues/640. 

I did an initial exploratory test with WP Desktop 5.0.1-beta2 and again with 5.0.0 (latest stable) and I got really stuck. 😕 ([3m52s](https://cloudup.com/cYrDAk9N4jf), [8m21s](https://cloudup.com/cb19VM-m1Uj))

Steps from my test:

1. Setup up a WordPress site.
1. Install, activate, and connect the Jetpack plugin.
1. Install and activate the Classic Editor plugin.
1. Create a few posts to test with.
1. Deactivate the Classic Editor plugin.
1. Install WP Desktop 5.0.0 .
1. Log in to the WP Desktop app using the Jetpack-connected account for the site.
1. Try to create a new post using the WP Desktop app.

**Result:** whenever I try to create or open a post using the block editor in the WP Desktop app, I am redirected to the web instead. This results in a stuck placeholder image in the app that cannot be resolved by closing/re-opening or by deleting/re-installing the WP Desktop app. 

![blank-editor-placeholder-screen-in-wpdesktop-5 0 1-beta2](https://user-images.githubusercontent.com/1119271/78404122-fac80200-75ba-11ea-88a5-0b78d17f720e.png)
<p style="clear:both;"><sup>Tested with WPDesktop 5.0.1-beta2 on macOS 10.14.6 on a site running WP 5.4 Jetpack 8.3 with and without the Classic Editor 1.5 plugin activated.</sup></p>

**Workarounds:** to get out of the stuck editor showing only the placeholder in the app, you can either activate the Classic Editor plugin on the site or you can delete `~/Library/Application\ Support/WordPress.com/settings.json` to be able to re-open the app but the editor problem will recur if you try to open or create a post with the block editor (or at least it did in my tests so far).

**Comment by designsimply** [comment]
I re-opened #640 with blocker status after testing a bit more. We need to fix 640 before we can address 813. 🙂

**Comment by formosattic** [comment]
Reported in 2890872-zen, and was able to reproduce issue, with version 5.1.0-beta1, the option to enable the block editor isn't there

![Screen Shot on 2020-04-21 at 12:13:08](https://user-images.githubusercontent.com/15944096/79824711-7a323f80-83c9-11ea-8ba2-66bce1ee5040.png)

**Comment by skobylite** [comment]
Anyone still having this issue in 5.2?

**Comment by erricgunawan** [comment]
Got a report in 3090569-zen

> When I open a new post, nor when I edit a post there is no info on the new editor (Gutenberg) offered nor displayed. When I am on a web browser, I don't have this problem. I have been using Gutenberg, that is what is frustrating.

> Today I was on the WordPress Desktop app when the try the new editor suddenly appeared while I was modifying a post. I hit link and a new window opened in classic editor mode.

Happened on macOS Catalina 10.15.5 and WordPress Desktop App version 5.2.0.

**Comment by rachelmcr** [comment]
I ran into this today while testing WPDesktop 6.0.1-beta2 on macOS 10.15.5 with a Jetpack site that uses SSO.

I get the classic editor, and when I click the prompt to enable the block editor nothing happens:

![wpdesktop-gutenbergprompt](https://user-images.githubusercontent.com/8658164/87554281-d85e7e00-c6ab-11ea-9271-6498054e54db.gif)

I don't see any useful information in the dev console or in the app logs.

**Comment by nsakaimbo** [comment]
> I get the classic editor, and when I click the prompt to enable the block editor nothing happens:

@creativecoder Not sure if you might be able to help point in the right direction here. Is it possible the "Try the new block editor" button is disabled in the desktop app somehow? Any pointers on where to look?

**Comment by snowmads** [comment]
I have a user in 3143751-zen reporting this too - SSO is enabled and they can't enable the block editor in desktop app for mac OS 10.15.5. Asking what version of the app they're using, and letting them know it may be a bug.

**Comment by creativecoder** [comment]
@nsakaimbo Could be related to the `isClassicEditorForced` selector and the [conditional for the desktop app](https://github.com/Automattic/wp-calypso/blob/8be25f257628f5f62c3c49702c8641e2ecd9cea0/client/state/selectors/is-classic-editor-forced.js#L25) there.

There's a series of checks in the `is_jetpack_site_eligible_for_gutenframe` function in wpcom, including Jetpack version and a list of plugins known to be incompatible with the iframed editor -- if those don't pass `gutenberg-redirect` or `gutenberg-redirect-and-style` is returned for the `selectedEditor`.

But looking at the `render` functions of the opt-in components, the "Try the new block editor" button shouldn't show up at all if those checks don't pass. 😕

**Comment by rachelmcr** [comment]
I can confirm that the prompt to try the new block editor appeared in wpcom web for my Jetpack site, and worked there, so the issue was something specific to it not working in the app (as opposed to it being ineligible).

**Comment by nsakaimbo** [comment]
Update: as of latest Calypso master, the "Try the new block editor" button is broken for Simple sites as well, and in addition is rendered incorrectly:

<p align="center">
<img width="280" alt="Screen Shot 2020-07-22 at 4 04 00 PM" src="https://user-images.githubusercontent.com/8979548/88223321-415a6e80-cc35-11ea-844e-1a3cf06b737f.png">
</p>

@creativecoder Are you aware of any additional changes to the way element is rendered? (Again this is with latest Calypso master. The desktop app can now be built completely from within Calypso with `yarn run build-desktop`).

**Comment by nsakaimbo** [comment]
**Gutenberg Mini Preview Image**

There's a couple bugs at play here, but for starters, the image being displayed is supposed to be `/calypso/images/illustrations/gutenberg-mini.svg` [here](https://github.com/Automattic/wp-calypso/blob/faf7af6886dddd422a8e83d1d1fa15e3075596d9/client/post-editor/editor-gutenberg-opt-in-sidebar/index.jsx#L53), but for some strange reason the image isn't showing up at all. Looking at the history for this file indicates no recent changes to this logic.

I verified the image asset is in the same location in the desktop app bundle as it was in prior releases: `./Contents/Resources/app/calypso/public/images/illustrations/gutenberg-mini.svg`.

However, the console shows this error message:

```
GET http://127.0.0.1:41050/calypso/images/illustrations/gutenberg-mini.svg 404 (Not Found)
```

So for some reason the image is now being resolved against `calypso/images` and not `calypso/public/images` where it's been until present. 

Did something in the Calypso asset config change here?

FWIW a number of assets cannot be resolved in the desktop app:

<p align="center">
<img width="385" alt="Screen Shot 2020-07-22 at 5 35 27 PM" src="https://user-images.githubusercontent.com/8979548/88231232-da8f8200-cc41-11ea-93e7-215730430fec.png">
</p>

@sgomes just pinging you as I know you have a pretty good handle on how the various dependencies in the desktop app get resolved. Any insights/suggestions here?

I also verified the image asset is in the same location in the app bundle as it was in prior releases: `./Contents/Resources/app/calypso/public/images/illustrations/gutenberg-mini.svg`.

**Opt-in Dialog**

I added some debug statements to both the `EditorGutenbergOptInDialog` and `EditorGutenbergOptInSidebar` and can confirm that:

  * `optInEnabled` returns true in `EditorGutenbergOptInSidebar`, and therefore the sidebar component renders ([link](https://github.com/Automattic/wp-calypso/blob/faf7af6886dddd422a8e83d1d1fa15e3075596d9/client/post-editor/editor-gutenberg-opt-in-sidebar/index.jsx#L41-L57)). However:
  * The [click handler](https://github.com/Automattic/wp-calypso/blob/faf7af6886dddd422a8e83d1d1fa15e3075596d9/client/post-editor/editor-gutenberg-opt-in-sidebar/index.jsx#L50) attached to the `EditorGutenbergOptInSidebar` somehow isn't registering at all, therefore  the logic in `EditorGutenbergOptInDialog` doesn't even have a chance to fire (i.e. the opt in dialog's render method doesn't fire at all, indicating a break before the logic even gets [to this point](https://github.com/Automattic/wp-calypso/blob/faf7af6886dddd422a8e83d1d1fa15e3075596d9/client/post-editor/editor-gutenberg-opt-in-dialog/index.jsx#L67)).

**Comment by sgomes** [comment]
@nsakaimbo: I'm not aware of any changes to asset serving configuration, sorry 😕 Perhaps someone in @Automattic/team-calypso knows more?

**Comment by nsakaimbo** [comment]
@creativecoder I did some git bisecting and was able to determine that the breaking change was introduced by commit Automattic/wp-calypso@680cc77f39400aab3e090c373630c8dd0f1887cd (PR Automattic/wp-calypso#43846).

I verified that things work as expected in the commit immediately preceding this one (Automattic/wp-calypso@ 84afbc3d24220efb04f8201c014dd6bc619ae31b).

Seems like while the flag was necessary to hide the deprecation notice in the desktop app (so e2e tests would continue to work), but somehow this also makes the "Try the new block editor" button non-functional.

Could you please help take a look?

**Comment by nsakaimbo** [comment]
_Update_: the missing image is actually isolated to the wp-desktop package in the Calypso monorepo, which isn't live yet - it's a non-issue for the time being.

**Comment by nsakaimbo** [comment]
Fixed by Automattic/wp-calypso#44490.
